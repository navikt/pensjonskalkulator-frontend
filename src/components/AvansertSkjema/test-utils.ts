export function getPreviousMonth(): string {
  const today = new Date()
  const currentMonth = parseInt(
    today.toLocaleDateString('en-CA').slice(5, 7),
    10
  ) // Extracts the month from the date string
  const previousMonth = currentMonth - 1
  return previousMonth.toString()
}
