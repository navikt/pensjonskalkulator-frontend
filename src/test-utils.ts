import renderer from 'react-test-renderer'

import { render } from '@testing-library/react'

export function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON()
  expect(result).toBeDefined()
  expect(result).not.toBeInstanceOf(Array)
  return result as renderer.ReactTestRendererJSON
}

const customRender = (ui: React.ReactElement, options = {}) =>
  render(ui, {
    // wrap provider(s) here if needed
    wrapper: ({ children }) => children,
    ...options,
  })

export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
// override render export
export { customRender as render }
