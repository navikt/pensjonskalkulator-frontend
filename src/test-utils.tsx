import React, { PropsWithChildren } from 'react'
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'
import { MemoryRouter } from 'react-router-dom'

import { PreloadedState, createListenerMiddleware } from '@reduxjs/toolkit'
import { render, RenderOptions } from '@testing-library/react'

import { getTranslation_test } from '@/utils/__tests__/test-translations'

import { createUttaksalderListener } from './state/listeners/uttaksalderListener'
import {
  setupStore,
  RootState,
  AppStore,
  AppStartListening,
} from './state/store'
import { getTranslation_nb } from './translations/nb'

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>
  store?: AppStore
  hasRouter?: boolean
}

export const swallowErrors = (testFn: () => void) => {
  const error = console.error
  console.error = () => {}
  testFn()
  console.error = error
}

export const swallowErrorsAsync = async (testFn: () => Promise<void>) => {
  const cache = console.error
  console.error = () => {}
  await testFn()
  console.error = cache
}

function generateMockedTranslations() {
  const nbTranslations: Record<string, string> = getTranslation_nb()
  const testTranslations: Record<string, string> = getTranslation_test()
  const translationsInput = { ...nbTranslations, ...testTranslations }
  const translations: Record<string, string> = {}

  for (const key in translationsInput) {
    if (
      /<(?=.*? .*?\/ ?>|br|hr|input|!--|wbr)[a-z]+.*?>|<([a-z]+).*?<\/\1>|{[^}]+}/i.test(
        translationsInput[key]
      )
    ) {
      // for html or chunks keys: results in 'my_key' : 'my_key with some <html> or {chunk}'
      translations[key] = translationsInput[key]
    } else {
      // results in 'my_key' : 'my_key'
      translations[key] = key
    }
  }
  return translations
}

// Return an object with the store and all of RTL's query functions
export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    store = setupStore(preloadedState, true),
    hasRouter = true,
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {
  function Wrapper({ children }: PropsWithChildren<unknown>): JSX.Element {
    return (
      <Provider store={store}>
        <IntlProvider locale="nb" messages={generateMockedTranslations()}>
          {hasRouter ? <MemoryRouter>{children}</MemoryRouter> : children}
        </IntlProvider>
      </Provider>
    )
  }

  const listenerMiddleware = createListenerMiddleware()
  createUttaksalderListener(
    listenerMiddleware.startListening as AppStartListening
  )

  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}

export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'

export { renderWithProviders as render }
