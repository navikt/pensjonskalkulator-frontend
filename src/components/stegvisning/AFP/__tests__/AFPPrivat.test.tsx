import { describe, it, vi } from 'vitest'

import { AFPPrivat } from '..'
import { screen, render, waitFor, userEvent } from '@/test-utils'

const navigateMock = vi.fn()
vi.mock(import('react-router'), async (importOriginal) => {
  const actual = await importOriginal()
  return {
    ...actual,
    useNavigate: () => navigateMock,
  }
})

describe('stegvisning - AFP - født før 1963 og og fylt 67 år, eller født før 1963 og har vedtak om alderspensjon', () => {
  const onCancelMock = vi.fn()
  const onPreviousMock = vi.fn()
  const onNextMock = vi.fn()

  it('rendrer slik den skal når afp ikke er oppgitt', async () => {
    const user = userEvent.setup()
    const result = render(
      <AFPPrivat
        previousAfp={null}
        onCancel={onCancelMock}
        onPrevious={onPreviousMock}
        onNext={onNextMock}
      />
    )
    expect(screen.getByRole('heading', { level: 2 })).toHaveTextContent(
      'stegvisning.afpPrivat.title'
    )

    expect(
      screen.queryByTestId('om_livsvarig_AFP_i_privat_sektor')
    ).toBeVisible()

    const radioButtons = await screen.findAllByRole('radio')
    await waitFor(() => {
      expect(radioButtons).toHaveLength(2)
      expect(radioButtons[0]).not.toBeChecked()
      expect(radioButtons[1]).not.toBeChecked()
      expect(result.asFragment()).toMatchSnapshot()
    })
  })

  it('rendrer slik den skal når afp er oppgitt', async () => {
    const result = render(
      <AFPPrivat
        previousAfp="nei"
        onCancel={onCancelMock}
        onPrevious={onPreviousMock}
        onNext={onNextMock}
      />
    )
    expect(screen.getByRole('heading', { level: 2 })).toHaveTextContent(
      'stegvisning.afpPrivat.title'
    )
    const radioButtons = screen.getAllByRole('radio')
    await waitFor(() => {
      expect(radioButtons).toHaveLength(2)
      expect(radioButtons[0]).not.toBeChecked()
      expect(radioButtons[1]).toBeChecked()
      expect(result.asFragment()).toMatchSnapshot()
    })
  })

  it('validerer, viser feilmelding, fjerner feilmelding og kaller onNext når brukeren klikker på Neste', async () => {
    const user = userEvent.setup()
    render(
      <AFPPrivat
        previousAfp={null}
        onCancel={onCancelMock}
        onPrevious={onPreviousMock}
        onNext={onNextMock}
      />
    )
    const radioButtons = screen.getAllByRole('radio')

    await user.click(screen.getByText('stegvisning.neste'))

    waitFor(() => {
      expect(
        screen.getByText('stegvisning.afpPrivat.validation_error')
      ).toBeInTheDocument()
      expect(onNextMock).not.toHaveBeenCalled()
    })

    await user.click(radioButtons[0])

    expect(
      screen.queryByText('stegvisning.afpPrivat.validation_error')
    ).not.toBeInTheDocument()

    await user.click(screen.getByText('stegvisning.neste'))

    waitFor(() => {
      expect(onNextMock).toHaveBeenCalled()
    })
  })

  it('kaller onNext når brukeren klikker på Neste', async () => {
    const user = userEvent.setup()
    render(
      <AFPPrivat
        previousAfp={null}
        onCancel={onCancelMock}
        onPrevious={onPreviousMock}
        onNext={onNextMock}
      />
    )
    const radioButtons = screen.getAllByRole('radio')
    await user.click(radioButtons[0])
    await user.click(screen.getByText('stegvisning.neste'))
    waitFor(() => {
      expect(onNextMock).toHaveBeenCalled()
    })
  })

  it('kaller onPrevious når brukeren klikker på Tilbake', async () => {
    const user = userEvent.setup()
    render(
      <AFPPrivat
        previousAfp="ja_privat"
        onCancel={onCancelMock}
        onPrevious={onPreviousMock}
        onNext={onNextMock}
      />
    )
    await user.click(screen.getByText('stegvisning.tilbake'))
    expect(onPreviousMock).toHaveBeenCalled()
  })

  it('kaller onCancelMock når brukeren klikker på Avbryt', async () => {
    const user = userEvent.setup()
    render(
      <AFPPrivat
        previousAfp="ja_privat"
        onCancel={onCancelMock}
        onPrevious={onPreviousMock}
        onNext={onNextMock}
      />
    )

    await user.click(screen.getByText('stegvisning.avbryt'))

    expect(screen.getByText('stegvisning.avbryt')).toBeInTheDocument()
    await user.click(screen.getByText('stegvisning.avbryt'))
    waitFor(() => {
      expect(onCancelMock).toHaveBeenCalled()
    })
  })

  it('viser ikke avbryt knapp når onCancel ikke er definert', async () => {
    render(
      <AFPPrivat
        previousAfp={null}
        onCancel={undefined}
        onPrevious={onPreviousMock}
        onNext={onNextMock}
      />
    )
    expect(screen.queryByText('stegvisning.avbryt')).not.toBeInTheDocument()
  })
})
