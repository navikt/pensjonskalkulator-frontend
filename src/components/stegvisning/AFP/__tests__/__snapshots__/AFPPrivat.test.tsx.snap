// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`stegvisning - AFP - født før 1963 og og fylt 67 år, eller født før 1963 og har vedtak om alderspensjon > rendrer slik den skal når afp er oppgitt 1`] = `
<DocumentFragment>
  <section
    class="_section_67454f _section__largePadding_67454f _section__marginBotton_67454f"
    data-testid="afp-privat"
  >
    <form>
      <h2
        class="navds-heading navds-heading--medium navds-typo--spacing"
      >
        stegvisning.afpPrivat.title
      </h2>
      <p
        class="navds-body-long navds-body-long--large"
      >
        stegvisning.afp.ingress
      </p>
      <div
        class="navds-read-more navds-read-more--medium _readmorePrivat_10a2c6"
        data-volume="low"
      >
        <button
          aria-expanded="false"
          class="navds-read-more__button navds-body-short"
          data-state="closed"
          data-testid="readmore"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="navds-read-more__expand-icon"
            fill="none"
            focusable="false"
            height="1em"
            role="img"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M5.97 9.47a.75.75 0 0 1 1.06 0L12 14.44l4.97-4.97a.75.75 0 1 1 1.06 1.06l-5.5 5.5a.75.75 0 0 1-1.06 0l-5.5-5.5a.75.75 0 0 1 0-1.06"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <span>
            stegvisning.afp.readmore_privat_title
          </span>
        </button>
        <div
          aria-hidden="true"
          class="navds-read-more__content navds-read-more__content--closed navds-body-long navds-body-long--medium"
          data-state="closed"
        >
          stegvisning.afp.readmore_privat_list_title
          <ul
            class="_list_10a2c6"
          >
            <li>
              stegvisning.afp.readmore_privat_list_item1
            </li>
            <li>
              stegvisning.afp.readmore_privat_list_item2
            </li>
            <li>
              stegvisning.afp.readmore_privat_list_item3
            </li>
            <li>
              stegvisning.afp.readmore_privat_list_item4
            </li>
          </ul>
          Enkelte arbeidsgivere i privat sektor kan ha avtaler i offentlig sektor. Sjekk hos arbeidsgiveren din hva som gjelder for deg. Les om vilkårene til 
          <a
            class="navds-link navds-link--action navds-link--inline-text"
            href="https://www.afp.no"
            target="_blank"
          >
            AFP i privat sektor på afp.no
            <svg
              aria-labelledby="title-rb"
              fill="none"
              focusable="false"
              height="1.25rem"
              role="img"
              viewBox="0 0 24 24"
              width="1.25rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title
                id="title-rb"
              >
                application.global.external_link
              </title>
              <path
                clip-rule="evenodd"
                d="M20.532 3.471A.75.75 0 0 1 20.75 4v7.5a.75.75 0 0 1-1.5 0V5.81l-8.72 8.72a.75.75 0 1 1-1.06-1.06l8.72-8.72H12.5a.75.75 0 0 1 0-1.5H20c.206 0 .393.083.529.218l.001.002zM4.75 9A.25.25 0 0 1 5 8.75h7a.75.75 0 0 0 0-1.5H5A1.75 1.75 0 0 0 3.25 9v10c0 .966.784 1.75 1.75 1.75h10A1.75 1.75 0 0 0 16.75 19v-7a.75.75 0 0 0-1.5 0v7a.25.25 0 0 1-.25.25H5a.25.25 0 0 1-.25-.25z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </a>
        </div>
      </div>
      <fieldset
        aria-required="true"
        class="_radiogroup_10a2c6 navds-radio-group navds-radio-group--medium navds-fieldset navds-fieldset--medium"
        role="radiogroup"
      >
        <legend
          class="navds-fieldset__legend navds-label"
        >
          stegvisning.afpPrivat.radio_label
        </legend>
        <div
          class="navds-radio-buttons"
        >
          <div
            class="navds-radio navds-radio--medium"
          >
            <input
              aria-labelledby="rf"
              class="navds-radio__input"
              id="radio-re"
              name="afp"
              type="radio"
              value="ja_privat"
            />
            <label
              class="navds-radio__label"
              for="radio-re"
            >
              <span
                class="navds-radio__content"
              >
                <span
                  aria-hidden="true"
                  class="navds-body-short navds-body-short--medium"
                  id="rf"
                >
                  stegvisning.radio_ja
                </span>
              </span>
            </label>
          </div>
          <div
            class="navds-radio navds-radio--medium"
          >
            <input
              aria-labelledby="ri"
              checked=""
              class="navds-radio__input"
              id="radio-rh"
              name="afp"
              type="radio"
              value="nei"
            />
            <label
              class="navds-radio__label"
              for="radio-rh"
            >
              <span
                class="navds-radio__content"
              >
                <span
                  aria-hidden="true"
                  class="navds-body-short navds-body-short--medium"
                  id="ri"
                >
                  stegvisning.radio_nei
                </span>
              </span>
            </label>
          </div>
        </div>
        <div
          aria-live="polite"
          aria-relevant="additions removals"
          class="navds-fieldset__error"
          id="fieldset-error-rd"
        />
      </fieldset>
      <button
        class="_button_10a2c6 navds-button navds-button--primary navds-button--medium"
        type="submit"
      >
        <span
          class="navds-label"
        >
          stegvisning.neste
        </span>
      </button>
      <button
        class="_button_10a2c6 navds-button navds-button--secondary navds-button--medium"
        type="button"
      >
        <span
          class="navds-label"
        >
          stegvisning.tilbake
        </span>
      </button>
      <button
        class="_button_10a2c6 navds-button navds-button--tertiary navds-button--medium"
        type="button"
      >
        <span
          class="navds-label"
        >
          stegvisning.avbryt
        </span>
      </button>
    </form>
  </section>
</DocumentFragment>
`;

exports[`stegvisning - AFP - født før 1963 og og fylt 67 år, eller født før 1963 og har vedtak om alderspensjon > rendrer slik den skal når afp ikke er oppgitt 1`] = `
<DocumentFragment>
  <section
    class="_section_67454f _section__largePadding_67454f _section__marginBotton_67454f"
    data-testid="afp-privat"
  >
    <form>
      <h2
        class="navds-heading navds-heading--medium navds-typo--spacing"
      >
        stegvisning.afpPrivat.title
      </h2>
      <p
        class="navds-body-long navds-body-long--large"
      >
        stegvisning.afp.ingress
      </p>
      <div
        class="navds-read-more navds-read-more--medium _readmorePrivat_10a2c6 navds-read-more--open"
        data-volume="low"
      >
        <button
          aria-expanded="true"
          class="navds-read-more__button navds-body-short"
          data-state="open"
          data-testid="om_livsvarig_AFP_i_privat_sektor"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="navds-read-more__expand-icon"
            fill="none"
            focusable="false"
            height="1em"
            role="img"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M5.97 9.47a.75.75 0 0 1 1.06 0L12 14.44l4.97-4.97a.75.75 0 1 1 1.06 1.06l-5.5 5.5a.75.75 0 0 1-1.06 0l-5.5-5.5a.75.75 0 0 1 0-1.06"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <span>
            Om livsvarig AFP i privat sektor
          </span>
        </button>
        <div
          aria-hidden="false"
          class="navds-read-more__content navds-body-long navds-body-long--medium"
          data-state="open"
        >
          <p>
            Lorem
          </p>
        </div>
      </div>
      <fieldset
        aria-required="true"
        class="_radiogroup_10a2c6 navds-radio-group navds-radio-group--medium navds-fieldset navds-fieldset--medium"
        role="radiogroup"
      >
        <legend
          class="navds-fieldset__legend navds-label"
        >
          stegvisning.afpPrivat.radio_label
        </legend>
        <div
          class="navds-radio-buttons"
        >
          <div
            class="navds-radio navds-radio--medium"
          >
            <input
              aria-labelledby="r5"
              class="navds-radio__input"
              id="radio-r4"
              name="afp"
              type="radio"
              value="ja_privat"
            />
            <label
              class="navds-radio__label"
              for="radio-r4"
            >
              <span
                class="navds-radio__content"
              >
                <span
                  aria-hidden="true"
                  class="navds-body-short navds-body-short--medium"
                  id="r5"
                >
                  stegvisning.radio_ja
                </span>
              </span>
            </label>
          </div>
          <div
            class="navds-radio navds-radio--medium"
          >
            <input
              aria-labelledby="r8"
              class="navds-radio__input"
              id="radio-r7"
              name="afp"
              type="radio"
              value="nei"
            />
            <label
              class="navds-radio__label"
              for="radio-r7"
            >
              <span
                class="navds-radio__content"
              >
                <span
                  aria-hidden="true"
                  class="navds-body-short navds-body-short--medium"
                  id="r8"
                >
                  stegvisning.radio_nei
                </span>
              </span>
            </label>
          </div>
        </div>
        <div
          aria-live="polite"
          aria-relevant="additions removals"
          class="navds-fieldset__error"
          id="fieldset-error-r3"
        />
      </fieldset>
      <button
        class="_button_10a2c6 navds-button navds-button--primary navds-button--medium"
        type="submit"
      >
        <span
          class="navds-label"
        >
          stegvisning.neste
        </span>
      </button>
      <button
        class="_button_10a2c6 navds-button navds-button--secondary navds-button--medium"
        type="button"
      >
        <span
          class="navds-label"
        >
          stegvisning.tilbake
        </span>
      </button>
      <button
        class="_button_10a2c6 navds-button navds-button--tertiary navds-button--medium"
        type="button"
      >
        <span
          class="navds-label"
        >
          stegvisning.avbryt
        </span>
      </button>
    </form>
  </section>
</DocumentFragment>
`;

exports[`stegvisning - AFP - født før 1963 og og fylt 67 år, eller født før 1963 og har vedtak om alderspensjon > rendrer slik den skal når afp ikke er oppgitt 2`] = `
<DocumentFragment>
  <section
    class="_section_67454f _section__largePadding_67454f _section__marginBotton_67454f"
    data-testid="afp-privat"
  >
    <form>
      <h2
        class="navds-heading navds-heading--medium navds-typo--spacing"
      >
        stegvisning.afpPrivat.title
      </h2>
      <p
        class="navds-body-long navds-body-long--large"
      >
        stegvisning.afp.ingress
      </p>
      <div
        class="navds-read-more navds-read-more--medium _readmorePrivat_10a2c6 navds-read-more--open"
        data-volume="low"
      >
        <button
          aria-expanded="true"
          class="navds-read-more__button navds-body-short"
          data-state="open"
          data-testid="om_livsvarig_AFP_i_privat_sektor"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="navds-read-more__expand-icon"
            fill="none"
            focusable="false"
            height="1em"
            role="img"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M5.97 9.47a.75.75 0 0 1 1.06 0L12 14.44l4.97-4.97a.75.75 0 1 1 1.06 1.06l-5.5 5.5a.75.75 0 0 1-1.06 0l-5.5-5.5a.75.75 0 0 1 0-1.06"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <span>
            Om livsvarig AFP i privat sektor
          </span>
        </button>
        <div
          aria-hidden="false"
          class="navds-read-more__content navds-body-long navds-body-long--medium"
          data-state="open"
        >
          <p>
            Lorem
          </p>
        </div>
      </div>
      <fieldset
        aria-required="true"
        class="_radiogroup_10a2c6 navds-radio-group navds-radio-group--medium navds-fieldset navds-fieldset--medium"
        role="radiogroup"
      >
        <legend
          class="navds-fieldset__legend navds-label"
        >
          stegvisning.afpPrivat.radio_label
        </legend>
        <div
          class="navds-radio-buttons"
        >
          <div
            class="navds-radio navds-radio--medium"
          >
            <input
              aria-labelledby="r5"
              class="navds-radio__input"
              id="radio-r4"
              name="afp"
              type="radio"
              value="ja_privat"
            />
            <label
              class="navds-radio__label"
              for="radio-r4"
            >
              <span
                class="navds-radio__content"
              >
                <span
                  aria-hidden="true"
                  class="navds-body-short navds-body-short--medium"
                  id="r5"
                >
                  stegvisning.radio_ja
                </span>
              </span>
            </label>
          </div>
          <div
            class="navds-radio navds-radio--medium"
          >
            <input
              aria-labelledby="r8"
              class="navds-radio__input"
              id="radio-r7"
              name="afp"
              type="radio"
              value="nei"
            />
            <label
              class="navds-radio__label"
              for="radio-r7"
            >
              <span
                class="navds-radio__content"
              >
                <span
                  aria-hidden="true"
                  class="navds-body-short navds-body-short--medium"
                  id="r8"
                >
                  stegvisning.radio_nei
                </span>
              </span>
            </label>
          </div>
        </div>
        <div
          aria-live="polite"
          aria-relevant="additions removals"
          class="navds-fieldset__error"
          id="fieldset-error-r3"
        />
      </fieldset>
      <button
        class="_button_10a2c6 navds-button navds-button--primary navds-button--medium"
        type="submit"
      >
        <span
          class="navds-label"
        >
          stegvisning.neste
        </span>
      </button>
      <button
        class="_button_10a2c6 navds-button navds-button--secondary navds-button--medium"
        type="button"
      >
        <span
          class="navds-label"
        >
          stegvisning.tilbake
        </span>
      </button>
      <button
        class="_button_10a2c6 navds-button navds-button--tertiary navds-button--medium"
        type="button"
      >
        <span
          class="navds-label"
        >
          stegvisning.avbryt
        </span>
      </button>
    </form>
  </section>
</DocumentFragment>
`;

exports[`stegvisning - AFP - født før 1963 og og fylt 67 år, eller født før 1963 og har vedtak om alderspensjon > rendrer slik den skal når afp ikke er oppgitt 3`] = `
<DocumentFragment>
  <section
    class="_section_67454f _section__largePadding_67454f _section__marginBotton_67454f"
    data-testid="afp-privat"
  >
    <form>
      <h2
        class="navds-heading navds-heading--medium navds-typo--spacing"
      >
        stegvisning.afpPrivat.title
      </h2>
      <p
        class="navds-body-long navds-body-long--large"
      >
        stegvisning.afp.ingress
      </p>
      <div
        class="navds-read-more navds-read-more--medium _wrapper_11f63f _readmorePrivat_10a2c6 navds-read-more--open"
        data-volume="low"
      >
        <button
          aria-expanded="true"
          class="navds-read-more__button navds-body-short"
          data-state="open"
          data-testid="om_livsvarig_AFP_i_privat_sektor"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="navds-read-more__expand-icon"
            fill="none"
            focusable="false"
            height="1em"
            role="img"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M5.97 9.47a.75.75 0 0 1 1.06 0L12 14.44l4.97-4.97a.75.75 0 1 1 1.06 1.06l-5.5 5.5a.75.75 0 0 1-1.06 0l-5.5-5.5a.75.75 0 0 1 0-1.06"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <span>
            Om livsvarig AFP i privat sektor
          </span>
        </button>
        <div
          aria-hidden="false"
          class="navds-read-more__content navds-body-long navds-body-long--medium"
          data-state="open"
        >
          <p>
            Lorem
          </p>
        </div>
      </div>
      <fieldset
        aria-required="true"
        class="_radiogroup_10a2c6 navds-radio-group navds-radio-group--medium navds-fieldset navds-fieldset--medium"
        role="radiogroup"
      >
        <legend
          class="navds-fieldset__legend navds-label"
        >
          stegvisning.afpPrivat.radio_label
        </legend>
        <div
          class="navds-radio-buttons"
        >
          <div
            class="navds-radio navds-radio--medium"
          >
            <input
              aria-labelledby="r5"
              class="navds-radio__input"
              id="radio-r4"
              name="afp"
              type="radio"
              value="ja_privat"
            />
            <label
              class="navds-radio__label"
              for="radio-r4"
            >
              <span
                class="navds-radio__content"
              >
                <span
                  aria-hidden="true"
                  class="navds-body-short navds-body-short--medium"
                  id="r5"
                >
                  stegvisning.radio_ja
                </span>
              </span>
            </label>
          </div>
          <div
            class="navds-radio navds-radio--medium"
          >
            <input
              aria-labelledby="r8"
              class="navds-radio__input"
              id="radio-r7"
              name="afp"
              type="radio"
              value="nei"
            />
            <label
              class="navds-radio__label"
              for="radio-r7"
            >
              <span
                class="navds-radio__content"
              >
                <span
                  aria-hidden="true"
                  class="navds-body-short navds-body-short--medium"
                  id="r8"
                >
                  stegvisning.radio_nei
                </span>
              </span>
            </label>
          </div>
        </div>
        <div
          aria-live="polite"
          aria-relevant="additions removals"
          class="navds-fieldset__error"
          id="fieldset-error-r3"
        />
      </fieldset>
      <button
        class="_button_10a2c6 navds-button navds-button--primary navds-button--medium"
        type="submit"
      >
        <span
          class="navds-label"
        >
          stegvisning.neste
        </span>
      </button>
      <button
        class="_button_10a2c6 navds-button navds-button--secondary navds-button--medium"
        type="button"
      >
        <span
          class="navds-label"
        >
          stegvisning.tilbake
        </span>
      </button>
      <button
        class="_button_10a2c6 navds-button navds-button--tertiary navds-button--medium"
        type="button"
      >
        <span
          class="navds-label"
        >
          stegvisning.avbryt
        </span>
      </button>
    </form>
  </section>
</DocumentFragment>
`;
